libcontainer/cgroups/utils.go

```diff
@@ -34,7 +34,9 @@ func IsCgroup2UnifiedMode() bool {
 	isUnifiedOnce.Do(func() {
 		var st unix.Statfs_t
 		if err := unix.Statfs(unifiedMountpoint, &st); err != nil {
-			panic("cannot statfs cgroup root")
+			logrus.WithError(err).Debug("cannot statfs cgroup root")
```

Comment:
1. It should be `Info` not `Debug` I think as it's pretty major.
2. Amend the message to say "assuming cgroup v1".

Reply:
3. Maybe add a TODO item to change the default to cgroupv2 somewhere in 2024 :)

Reply:
yeah, this is partially why I am leaning towards changing the signature to return an `error`, otherwise users will get this message 100% of the time exec'ing to a container via buildkit with rootless.

Reply:
Does something like this seem more reasonable to you?
```
		err := unix.Statfs(unifiedMountpoint, &st)
		if err != nil && os.IsNotExist(err) {
			// /sys/fs/cgroup not found, likely rootless
			logrus.WithError(err).Debugf("%s missing, assuming cgroup v1", unifiedMountpoint)
			isUnified = false
			return
		} else if err != nil {
			panic(fmt.Sprintf("cannot statfs cgroup root: %s", err))
		}
```
This would preserve the panic behavior for all error scenarios except where the `/sys/fs/cgroup` path simply does not exist (which is a normal situation under rootless)

Reply:
Also better to panic when `os.IsNotExist && !system.RunningInUserNS()`

Reply:
updated the check to only ignore (w/ debug msg) errors for NotExists while RunningInUserNS